### PR TITLE
Issue 1165 : nccl + torch.cuda not available

### LINKED ISSUE #1165: idist initialize with nccl backend without GPU
## 🐛 Bug description

Call `idist.initialize("nccl", ...)` without GPU (but pytorch providing `nccl`) raises an error 
```commandline
Floating point exception
```
Exception is thrown in barrier following `dist.initialize`.

Case `nccl` + no GPU should raise an explicit exception.

## Environment

 - PyTorch Version (e.g., 1.4): 1.5
 - Ignite Version (e.g., 0.3.0): 0.4
 - OS (e.g., Linux): Linux
 - How you installed Ignite (`conda`, `pip`, source): conda
 - Python version: 3.7
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by sdesrozis**
Fixes #1165 

Description:

Raise an exception if `nccl` and `torch.cuda` is not available.

Check list:
* [x] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/distributed/comp_models/native.py [hunk 1]
```diff
@@ -74,6 +74,9 @@ def _create_from_backend(self, backend, timeout=None, **kwargs):
             if timeout is not None:
                 init_pg_kwargs["timeout"] = timeout
 
+            if backend == dist.Backend.NCCL and not torch.cuda.is_available():
+                raise RuntimeError("nccl backend is required but torch.cuda is not available")
```
Comment (vfdev-5) [inline ignite/distributed/comp_models/native.py:78]:
```suggestion
                raise RuntimeError("Nccl backend is required but no cuda capable devices")
```

**Comment by sdesrozis** [comment]
Yes I have to do that to close this PR.

**Comment by vfdev-5** [comment]
> Yes I have to do that to close this PR.
> EDIT : not so easy to do it in pytest https://discuss.pytorch.org/t/disable-cuda-within-one-single-pytest-test/34037
> @vfdev-5 do you have a smart way in mind to test this ?

@sdesrozis here is what you can do on a system with available GPU and torch dist NCCL
```python
from unittest.mock import patch

import torch
import torch.distributed as dist

import pytest


def test_cuda_and_nccl():    
    assert torch.cuda.is_available()
    assert dist.is_nccl_available()

    
@pytest.fixture()
def mock_gpu_is_not_available():

    with patch("torch.cuda") as mock_cuda:
        mock_cuda.is_available.return_value = False
        yield mock_cuda


def test_no_cuda_but_nccl(mock_gpu_is_not_available):
    assert (not torch.cuda.is_available())
    assert dist.is_nccl_available()
    
    
def test_cuda_and_nccl_2():
    assert torch.cuda.is_available()
    assert dist.is_nccl_available()


> pytest test.py
```

**Comment by vfdev-5** [comment]
Seems like does not work on GPUs:
- https://app.circleci.com/pipelines/github/pytorch/ignite/483/workflows/fa6cd65f-691e-46bc-a6af-b05ceb3ce481/jobs/922/steps

**Comment by sdesrozis** [comment]
Ok I check soon 😔

**Pull request merged**